src/util/browser.js

```diff
@@ -37,15 +39,22 @@ const exported = {
     },
 
     getImageData(img: CanvasImageSource, padding?: number = 0): ImageData {
-        const canvas = window.document.createElement('canvas');
+        const {width, height} = img;
+
+        if (!canvas) {
+            canvas = window.document.createElement('canvas');
+        }
+
         const context = canvas.getContext('2d');
         if (!context) {
             throw new Error('failed to create canvas 2d context');
         }
-        canvas.width = img.width;
-        canvas.height = img.height;
-        context.drawImage(img, 0, 0, img.width, img.height);
-        return context.getImageData(-padding, -padding, img.width + 2 * padding, img.height + 2 * padding);
+
+        canvas.width = width;
+        canvas.height = height;
```

Comment:
Do you know why this commit didn't work out? https://github.com/mapbox/mapbox-gl-js/pull/11564/commits/9d70bca3bd2e456f8fc0de58acc26e2bf3f86ce6
```
if (!canvas.width || canvas.width < width || !canvas.height || canvas.height < height) {
  canvas.width = width;
  canvas.height = height;
}
```
I'm concerned about these downsizing the canvas unnecessarily. Have you tried:
```
canvas.width = width > canvas.width ? width : canvas.width;
```